pulsar-client/src/main/java/org/apache/pulsar/client/impl/PatternTopicsConsumerImpl.java

```diff
@@ -36,11 +50,119 @@ public PatternTopicsConsumerImpl(Pattern topicsPattern,
                               CompletableFuture<Consumer> subscribeFuture) {
         super(client, conf, listenerExecutor, subscribeFuture);
         this.topicsPattern = topicsPattern;
+
+        if (this.namespaceName == null) {
+            this.namespaceName = getNameSpaceFromPattern(topicsPattern);
+        }
+        checkArgument(getNameSpaceFromPattern(topicsPattern).toString().equals(this.namespaceName.toString()));
+
+        this.topicsChangeListener = new PatternTopicsChangedListener();
+        scheduledExecutor = (ScheduledExecutorService)listenerExecutor;
+        scheduledExecutor.scheduleAtFixedRate(() -> recheckTopics(), 60, 30, TimeUnit.SECONDS);
+    }
+
+    public static NamespaceName getNameSpaceFromPattern(Pattern pattern) {
+        return TopicName.get(pattern.pattern()).getNamespaceObject();
```

Comment:
What if the pattern was missing the `/` or if there there was a pattern in the namespace name? Is this already handled somewhere else?

Reply:
Thanks, If there was missing part, in class `TopicName` will throw IllegalArgumentException. If there was regex contains in namespace name, it will get no topics to sub later,  or in TopicsConsumerImpl `topicNamesValid` will return false.